### PR TITLE
feat: add udf reverse()

### LINKED ISSUE #786: built-in function: Support reverse() function
**Is your feature request related to a problem? Please describe.**

reverse(str) - Returns the reversed given string.

**Syntax**
```
REVERSE(string)
```

```sql
SELECT reverse('abc');

-- output cba
```

**Describe the solution you'd like**

We do provide a built-in function development guide for you. 
Please check [OpenMLDB Build-In Function Develop Guide](https://github.com/4paradigm/OpenMLDB/blob/main/docs/en/built_in_function_develop_guide.md) or the [Chinese version](https://github.com/4paradigm/OpenMLDB/blob/main/docs/cn/built_in_function_develop_guide.md) for help.

Please make sure you:
- [x] Add built-in C++ function in src/udf/udf.h and src/udf/udf.cc
- [x] Register function to default library from function `void DefaultUdfLibrary::InitStringUdf()` in src/udf/default_udf_library.cc.
- [x] Add related unit test in src/codegen/udf_ir_builder_test.cc
- [x] Documenting function 

If you have any questions or concerns, please let us know. We appreciate every piece of code, comment.
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by nautaa**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
add udf reverse()

* **What is the current behavior?** (You can also link to an open issue here)

close #786 

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by lumianph** [comment]
@nautaa thank you for your contribution 👍 
@jingchen2222 @aceforeverd  please help review this PR

**Review by jingchen2222 — APPROVED** [review]
@nautaa Thanks for your contribution. Really appriciative.

File: hybridse/src/udf/udf.cc [hunk 1]
```diff
@@ -836,6 +836,19 @@ void ucase(codec::StringRef *str, codec::StringRef *output, bool *is_null_ptr) {
     *is_null_ptr = false;
 }
 
+void reverse(codec::StringRef *str, codec::StringRef *output, bool *is_null_ptr) {
+    if (str == nullptr || str->size_ == 0 || output == nullptr || is_null_ptr == nullptr) {
```
Comment (aceforeverd) [inline hybridse/src/udf/udf.cc]: Returned strref not modified if str size == 0 ?

Comment (nautaa) [reply hybridse/src/udf/udf.cc]: Done.

**Review by aceforeverd — APPROVED** [review]

**Comment by jingchen2222** [comment]
@nautaa Thanks again for your contribution.

**Comment by lumianph** [comment]
[community pr/issue tracking](https://trello.com/c/S4D8P57j/17-community-pr-issue-tracking)

**Pull request merged**